assets/js/modules/analytics/datastore/profiles.js

```diff
@@ -0,0 +1,358 @@
+
+			return {
+				...state,
+				isFetchingCreateProfile: {
+					...state.isFetchingCreateProfile,
+					[ `${ accountID }::${ propertyID }` ]: false,
+				},
+				profiles: {
+					...state.profiles,
+					[ `${ accountID }::${ propertyID }` ]: [
+						...( state.profiles[ `${ accountID }::${ propertyID }` ] || [] ),
+						profile,
+					],
+				},
+			};
+		}
+
+		case RECEIVE_CREATE_PROFILE_FAILED: {
+			const { accountID, error, propertyID } = payload;
+
+			return {
+				...state,
+				error,
+				isFetchingCreateProfile: {
+					...state.isFetchingCreateProfile,
+					[ `${ accountID }::${ propertyID }` ]: false,
+				},
+			};
+		}
+
+		case RECEIVE_PROFILES: {
+			const { profiles } = payload;
+
+			return {
+				...state,
+				profiles: {
+					...state.profiles,
+					...groupBy( profiles, ( { accountId, webPropertyId } ) => `${ accountId }::${ webPropertyId }` ), // Capitalization rule exception: `accountId` and `webPropertyId` are properties of an API returned value.
+				},
+			};
+		}
+
+		case RECEIVE_PROFILES_COMPLETED: {
+			const { accountID, propertyID } = payload;
+
+			return {
+				...state,
+				isFetchingProfiles: {
+					...state.isFetchingProfiles,
+					[ `${ accountID }::${ propertyID }` ]: false,
+				},
+			};
+		}
+
+		case RECEIVE_PROFILES_FAILED: {
+			const { accountID, error, propertyID } = payload;
+
+			return {
+				...state,
+				error,
+				isFetchingProfiles: {
+					...state.isFetchingProfiles,
+					[ `${ accountID }::${ propertyID }` ]: false,
+				},
+			};
+		}
+
+		default: {
+			return { ...state };
+		}
+	}
+};
+
+export const resolvers = {
+	*getProfiles( accountID, propertyID ) {
+		try {
+			const profiles = yield actions.fetchProfiles( accountID, propertyID );
+
+			yield actions.receiveProfiles( profiles );
+
+			return actions.receiveProfilesCompleted( accountID, propertyID );
+		} catch ( error ) {
+			// TODO: Implement an error handler store or some kind of centralized
+			// place for error dispatch...
+			return actions.receiveProfilesFailed( { accountID, propertyID, error } );
+		}
+	},
+};
+
+export const selectors = {
+	/**
+	 * Get all Google Analytics profiles this user account+property has available.
+	 *
+	 * Returns an array of all profiles.
+	 *
+	 * Returns `undefined` if accounts have not yet loaded.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state      Data store's state.
+	 * @param {string} accountID  The Analytics Account ID to fetch profiles for.
+	 * @param {string} propertyID The Analytics Property ID to fetch profiles for.
+	 * @return {Array|undefined} An array of Analytics profiles; `undefined` if not loaded.
+	 */
+	getProfiles( state, accountID, propertyID ) {
+		invariant( accountID, 'accountID is required.' );
+		invariant( propertyID, 'propertyID is required.' );
+
+		const { profiles } = state;
+
+		if ( 'undefined' === typeof profiles || 'undefined' === typeof profiles[ `${ accountID }::${ propertyID }` ] ) {
+			return undefined;
+		}
```

Comment:
I think we can remove this clause altogether (like in `getProperties`), with your changes below this should already work as expected (return the array or `undefined`).